qiskit/visualization/gate_map.py

```diff
@@ -733,12 +733,15 @@ def plot_error_map(backend, figsize=(12, 9), show_title=True):
 
     num_qubits = config["n_qubits"]
 
-    # U2 error rates
+    # sx error rates
     single_gate_errors = [0] * num_qubits
     for gate in props["gates"]:
-        if gate["gate"] == "u2":
+        if gate["gate"] == "sx":
             _qubit = gate["qubits"][0]
-            single_gate_errors[_qubit] = gate["parameters"][0]["value"]
+            for param in gate["parameters"]:
+                if param["name"] == "gate_error":
+                    single_gate_errors[_qubit] = param["value"]
+                    break
```

Comment:
Are we able to 100% guarantee that there will be a parameter called `gate_error` in this case?  If not, should we throw an error if it's not present, or is it ok to silently ignore?

Reply:
Well, it is a required item in the old Qiskit specification, so I am hoping there is some guarantee it is there.  I am more hoping that we get rid of this parameters format and move to something more sane, e.g. https://github.com/Qiskit-Partners/qiskit-ibm/issues/85

Reply:
Yeah, definitely agree with what you've said in that linked issue, and that this should be far more abstract than it currently is.  While it stays in this awkward form, it might be safest to add an else clause:
```python
for param in gate["parameters"]:
    if param["name"] == "gate_error":
        single_gate_errors[_qubit] = param["value"]
        break
else:
    raise VisualizationError("Backend did not supply a 'gate_error' property for this 'sx' gate.")
```
so that the error isn't silently ignored for the time being (or whatever error message makes sense - I'm not very sure).

Otherwise, this change looks all fine to me, and checking through the parameters in a loop does seem much safer than assuming the 0th parameter will be the one we want.